hazelcast/src/main/java/com/hazelcast/internal/management/dto/AliasedDiscoveryConfigDTO.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.management.dto;
+
+import com.hazelcast.config.AliasedDiscoveryConfig;
+import com.hazelcast.config.AliasedDiscoveryConfigUtils;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.internal.json.JsonValue;
+import com.hazelcast.internal.management.JsonSerializable;
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static com.hazelcast.util.JsonUtil.fromJsonObject;
+import static com.hazelcast.util.JsonUtil.toJsonObject;
+import static com.hazelcast.util.MapUtil.isNullOrEmpty;
+
+/**
+ * A JSON representation of a
+ * {@link com.hazelcast.config.AliasedDiscoveryConfig} implementation.
+ *
+ * @param <C> configuration implementation type
+ */
+public class AliasedDiscoveryConfigDTO<C extends AliasedDiscoveryConfig<C>> implements JsonSerializable {
+
+    private String tag;
+    private C config;
+
+    public AliasedDiscoveryConfigDTO(String tag) {
+        this.tag = tag;
+    }
+
+    public AliasedDiscoveryConfigDTO(C config) {
+        this.config = config;
+    }
+
+    @Override
+    public JsonObject toJson() {
+        JsonObject root = new JsonObject()
+                .add("enabled", config.isEnabled())
+                .add("usePublicIp", config.isUsePublicIp());
+
+        if (!isNullOrEmpty(config.getProperties())) {
+            root.add("properties", toJsonObject(config.getProperties()));
+        }
+
+        return root;
+    }
+
+    @Override
+    public void fromJson(JsonObject json) {
+        config = AliasedDiscoveryConfigUtils.<C>newConfigFor(tag);
```

Comment:
Minor: no need for the explicit type.

Reply:
Yes, tried to remove that but for some reason I get `incompatible types: inference variable C has incompatible upper bounds com.hazelcast.config.AliasedDiscoveryConfig<C>,C`. My generics foo is not strong enough to solve that :)